### ISSUE TITLE
Odometry disconnect at the end of long chains

### ISSUE THREAD

**Issue by mc2922** [JuliaRobotics/Caesar.jl#382]
Odometry chains break towards the end of a longer chain? 

190 poses:
[ex2-1.pdf](https://github.com/JuliaRobotics/Caesar.jl/files/3531960/ex2-1.pdf)
[ex2-2.pdf](https://github.com/JuliaRobotics/Caesar.jl/files/3531961/ex2-2.pdf)

200 poses:
[ex1-1.pdf](https://github.com/JuliaRobotics/Caesar.jl/files/3531955/ex1-1.pdf)
[ex1-2.pdf](https://github.com/JuliaRobotics/Caesar.jl/files/3531956/ex1-2.pdf)

**Comment by dehann** [comment]
Depiction of the Bayes tree that associated with this issue.  The short branch has a prior, but that prior information is not properly distributed down the long sibling branch.

![IMG_1675](https://user-images.githubusercontent.com/6412556/63656854-a7a9e900-c767-11e9-84ba-cb9db5e5a192.jpg)

**Comment by dehann** [comment]
another anecdotal piece of information is that the problem also occurred on a single long branch.

**Comment by dehann** [comment]
From the photo above, the triangle is x169, square is x170.  When looking at the clique low down on right hand branch (x169), the following subgraph is obtained.  Note priors represent the downward belief messages during the final stages of the complete solve.  This is where the disconnect in odo seems to be occurring:

![Screenshot from 2019-09-02 12-33-06](https://user-images.githubusercontent.com/6412556/64126980-df271f80-cd7d-11e9-967b-295537cdba4b.png)

**Comment by dehann** [comment]
All these loose fragments should also be computed in parallel btw.

**Comment by dehann** [comment]
Ha, a little more progress.  By looking closer I found that odometry factor x168--x169 is working the wrong way round for some reason,
![Screenshot from 2019-09-02 23-35-02](https://user-images.githubusercontent.com/6412556/64142780-7a9aad80-cddb-11e9-9e27-53fc7feafad9.png)

 and then found this.  The order of fields x168 and x169 is important here, **but clearly wrong**:
![Screenshot from 2019-09-02 23-37-58](https://user-images.githubusercontent.com/6412556/64142644-e16b9700-cdda-11e9-9063-f233b335fdaa.png)

@GearsAD , do you have any ideas here -- think we spoke a little while back on this but I cannot recall what we decided to do about `fncargvID`?

Values in factor seem fine:
![Screenshot from 2019-09-02 23-35-31](https://user-images.githubusercontent.com/6412556/64142747-58a12b00-cddb-11e9-83b3-0e9ca9ab186f.png)

Oh, and the factor graph here `sfg31bd` stands for sub factor graph inside cliq 31 before the down solve is performed.
![Screenshot from 2019-09-02 23-44-33](https://user-images.githubusercontent.com/6412556/64142820-aa49b580-cddb-11e9-8ee3-bc2e9d6de302.png)

**Comment by dehann** [comment]
this does not seem to be present in the original factor graph:
![Screenshot from 2019-09-02 23-51-44](https://user-images.githubusercontent.com/6412556/64143123-b97d3300-cddc-11e9-8ca7-86f83920b8c3.png)

**Comment by dehann** [comment]
right, so part of the problem was that downward results were not being transferred from cliq sub graphs back to main dfg -- this was an oversight in the new CSM code.  However, the issue is not yet fully resolved since downward messages from root into the target clique might have inconsistencies, and I'm looking specifically at that now.

**Comment by dehann** [comment]
performance should be better with II v0.7.7 or greater.  Keep open for rigorous testing

**Comment by dehann** [comment]
need to retest -- currently on IIF v0.8.3 with many improvements since this issue was logged.

**Comment by dehann** [comment]
Will re-evaluate after JuliaRobotics/IncrementalInference.jl#579 is fixed
